File: backend/src/domain/entities/GlobalProfile.ts
import { Entity, PrimaryColumn, Column } from 'typeorm';

@Entity('global_profiles')
export class GlobalProfile {
  @PrimaryColumn({ name: 'user_id' })
  userId: number;

  @Column({ type: 'varchar', name: 'full_name', nullable: true })
  fullName: string;

  @Column({ type: 'varchar', nullable: true })
  phone: string;

  @Column({ type: 'varchar', name: 'avatar_url', nullable: true })
  avatarUrl: string;

  @Column({ type: 'text', nullable: true })
  bio: string;

  @Column({ type: 'date', nullable: true })
  dob: Date;

  @Column({ type: 'varchar', nullable: true })
  gender: string;

  @Column({ type: 'json', nullable: true })
  settings: any;
}

File: backend/src/main/java/com/service/backend/authmodule/domain/entity/GlobalProfile.java
package com.service.backend.authmodule.domain.entity;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.relational.core.mapping.Column;
import org.springframework.data.relational.core.mapping.Table;

import java.time.LocalDate;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Table("global_profiles")
public class GlobalProfile {

    @Id
    @Column("user_id")
    private Long userId;

    @Column("full_name")
    private String fullName;

    private String phone;

    @Column("avatar_url")
    private String avatarUrl;

    private String bio;

    private LocalDate dob;

    private String gender;

    private String settings; // JSON stored as String
}
